client/scripts/views/pages/profile/index.ts

```diff
@@ -244,16 +244,24 @@ const ProfilePage: m.Component<{ address: string, setIdentity?: boolean }, IProf
         const chainInfo = app.config.chains.getById(chain);
         if (chainInfo?.base === ChainBase.Substrate) {
           try {
-            // TODO: should we enforce specific chain checksums here?
-            decodeAddress(address);
-            vnode.state.account = {
-              profile: null,
-              chain,
-              address,
-              id: null,
-              name: null,
-              user_id: null,
-            };
+            const decodedAddress = decodeAddress(address);
```

Comment:
I would move this decode operation into the `!valid` branch, as it might be time consuming to run that WASM code, or even use the `AddressSwapper` instead of using it manually.